csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -60,13 +61,49 @@ public void Dispose()
 
     public class DisposableNamedOnnxValue : NamedOnnxValue, IDisposable
     {
-        protected IDisposable _nativeMemoryManager;
-        protected DisposableNamedOnnxValue(string name, Object value, IDisposable nativeMemoryManager)
+        private NativeMemoryHandler _nativeMemoryManager;
+        private DisposableNamedOnnxValue(string name, Object value, NativeMemoryHandler nativeMemoryManager)
             : base(name, value)
         {
             _nativeMemoryManager = nativeMemoryManager;
         }
 
+        /// <summary>
+        /// Overrides the base class method. Since the instance already has access to the 
+        /// underlying OrtValue handle (if this instance hasn't been disposed), it just assigns
+        /// that to the output onnxValue. With respect to pinnedMemoryHandle, it has no operation
+        /// to do, as this class doesn't maintain a managed buffer. It doesn't have to maintain it
+        /// as it already is associated with the object of interest (native OrtValue)
+        /// </summary>
+        /// <param name="onnxValue"></param>
+        /// <param name="pinnedMemoryHandle"></param>
+        /// <param name="disposeOnnxValueAfterUse"></param>
+        internal override void ToNativeOnnxValue(out IntPtr onnxValue, 
+                                                 out MemoryHandle pinnedMemoryHandle,
+                                                 out bool disposeOnnxValueAfterUse)
+        {
+            // Make sure that this instance hasn't been disposed yet
+            if (disposedValue)
+            {
+                throw new ObjectDisposedException(nameof(DisposableNamedOnnxValue),
+                                                  "This instance of DisposableNamedOnnxValue has already been disposed");
+            }
+
+            // If not already disposed, _nativeMemoryManager can only be null
+            // for Maps and SequenceTensors
+            if (_nativeMemoryManager == null)
+            {
+                throw new NotSupportedException("Use of Maps and SequenceTensors is not yet supported");
+            }
+
+            // 'disposeOnnxValueAfterUse' is always 'false' for DisposableNamedOnnxValue as the onus
+            // to dispose the onnxValue after use is on the user, not the internal caller
+            disposeOnnxValueAfterUse = false;
+
+            // Assign the onnxValue by querying this instance's NativeOnnxTensorMemory instance
+            onnxValue = _nativeMemoryManager.Handle;
```

Comment:
How about pinnedMemoryHandle? It's an out value, but not being set in method.

Reply:
Its is not required. DisposableNamedOnnxValue has no managed memory that is pinned and hence this doesn't need to get set in this override method. The base class method (NamedOnnxValue) needs to set it.

Reply:
For out values in a function, they should be set to some valid value at least  (even if it's not used later on in the execution flow). What will happen if the caller attempts to use pinnedMemoryHandle, but it's not set to a valid value?

Reply:
pinnedMemoryHandle is set to default for completeness' sake and strictly speaking the caller should not be trying to use pinnedMemoryHandle from a DisposableNamedOnnxValue instance...